python/ray/data/_internal/planner/plan_expression/expression_visitors.py

```diff
@@ -8,11 +8,51 @@
     DownloadExpr,
     Expr,
     LiteralExpr,
+    Operation,
     StarExpr,
     UDFExpr,
     UnaryExpr,
     _ExprVisitor,
 )
+from ray.util import log_once
+
+try:
+    from pyiceberg.expressions import (
+        And,
+        BooleanExpression,
+        EqualTo,
+        GreaterThan,
+        GreaterThanOrEqual,
+        In,
+        IsNull,
+        LessThan,
+        LessThanOrEqual,
+        Not,
+        NotEqualTo,
+        NotIn,
+        NotNull,
+        Or,
+    )
+
+    # Map Ray Data operations to Iceberg operations
+    RAY_DATA_OPERATION_TO_ICEBERG = {
+        Operation.EQ: EqualTo,
+        Operation.NE: NotEqualTo,
+        Operation.GT: GreaterThan,
+        Operation.GE: GreaterThanOrEqual,
+        Operation.LT: LessThan,
+        Operation.LE: LessThanOrEqual,
+        Operation.AND: And,
+        Operation.OR: Or,
+        Operation.IN: In,
+        Operation.NOT_IN: NotIn,
+        Operation.IS_NULL: IsNull,
+        Operation.IS_NOT_NULL: NotNull,
+        Operation.NOT: Not,
+    }
+except ImportError:
+    log_once("pyiceberg is not installed, some expression visitors will be unavailable")
```

Comment:
Will we always log this warning, even if the user isn't using Iceberg at all? If so, I think we should rewrite this to avoid that. I think it could be confusing